### PR TITLE
Upgrading the Elasticsearch module to use ECK 1.6.0 and ES 7.14

### PR DESCRIPTION
**Pull request by Avilir**
Signed-off-by: Avi Layani <alayani@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/elasticsearch.py [hunk 1]
```diff
@@ -53,21 +55,17 @@ def get_data_from_text_file(json_file):
             l.strip() for l in open(str(json_file), encoding="utf8", errors="ignore")
         ]
         log.info(f"String docs length: {len(docs)}")
-        doc_list = []
 
         for num, doc in enumerate(docs):
             try:
                 dict_doc = json.loads(doc)
-                doc_list += [dict_doc]
+                yield dict_doc
```
Comment (keesturam) [inline ocs_ci/ocs/elasticsearch.py:62]: Nice yield implementation @Avilir

File: ocs_ci/ocs/elasticsearch.py [hunk 2]
```diff
@@ -271,7 +278,8 @@ def cleanup(self):
         log.info("Deleting the es resource")
         self.ocp.delete(yaml_file=self.crd)
         log.info("Deleting the es project")
-        self.ns_obj.delete_project(project_name=self.namespace)
+        # self.ns_obj.delete_project(project_name=self.namespace)
+        self.ocp.delete(yaml_file=self.eck_file)
```
Comment (keesturam) [inline ocs_ci/ocs/elasticsearch.py:282]: As part of the cleanup, you should also delete the PVC created in line 216

**Review by keesturam — COMMENTED** [review]
great to see the changes & code optimization @Avilir

File: ocs_ci/ocs/elasticsearch.py (hunk 2 above)
Comment (Avilir) [reply ocs_ci/ocs/elasticsearch.py:282]: @keesturam PVC can not be deleted before the pod which use it deleted, and the pod is deleted on line 279 which delete in the end the all project.
when the project deleted, there is no PVC to delete, so in the end, the cleanup delete also the PVC.

**Review by keesturam — APPROVED** [review]
LGTM

**Comment by keesturam** [comment]
@Avilir can we also run a sanity run of all performance tests that needs elasticsearch. This is to make sure that the change doesn't affect any existing tests.

**Comment by Avilir** [comment]
> @Avilir can we also run a sanity run of all performance tests that needs elasticsearch. This is to make sure that the change doesn't affect any existing tests.

yes we can, but it will take a lot of time.
will try to run in parallel, one PR validation with SF, and one with FIO

**Review by OdedViner — APPROVED** [review]
LGTM

**Pull request merged**